sky/clouds/service_catalog/images/aws_utils/image_gen.py

```diff
@@ -0,0 +1,148 @@
+        f'--launch-permission "{{\\\"Add\\\": [{{\\\"Group\\\":\\\"all\\\"}}]}}" --region {region}'
+    )
+    subprocess.run(public_command, shell=True, check=True)
+    print(f"Made {image_id} public")
+
+def copy_image_and_make_public(target_region):
+    # Copy the AMI to the target region
+    copy_command = (
+        f"aws ec2 copy-image --source-region {args.region} "
+        f"--source-image-id {args.image_id} --region {target_region} "
+        f"--name 'skypilot-aws-{args.processor}-{args.os_type}-{time.time()}'  --output json")
+    print(copy_command)
+    result = subprocess.run(copy_command,
+                            shell=True,
+                            check=True,
+                            capture_output=True,
+                            text=True)
+    print(result.stdout)
+    new_image_id = json.loads(result.stdout)['ImageId']
+    print(f"Copied image to {target_region} with new image ID: {new_image_id}")
+
+    # Wait for the image to be available
+    print(f"Waiting for {new_image_id} to be available...")
+    wait_command = f"aws ec2 wait image-available --image-ids {new_image_id} --region {target_region}"
+    subprocess.run(wait_command, shell=True, check=True)
+
+    make_image_public(new_image_id, target_region)
+
+    return new_image_id
+
+
+def write_image_to_csv(image_id, region):
+    with open(args.output_csv, 'a', newline='', encoding='utf-8') as csvfile:
+        writer = csv.writer(csvfile)
+        row = [
+            f'skypilot:custom-{args.processor}-{args.os_type}',
+            region, args.os_type, args.os_version, image_id,
+            time.strftime('%Y%m%d'), args.base_image_id
+        ]
+        writer.writerow(row)
+    print(f"Wrote to CSV: {row}")
+
+
+def main():
+    # Create CSV file if it doesn't exist
+    if not os.path.exists(args.output_csv):
+        with open(args.output_csv, 'w', newline='') as csvfile:
+            writer = csv.writer(csvfile)
+            writer.writerow([
+                'Tag', 'Region', 'OS', 'OSVersion', 'ImageId', 'CreationDate',
+                'BaseImageId'
+            ])  # Header
+        print(f"No existing {args.output_csv} so created it.")
+
+    # Process source AMI
+    make_image_public(args.image_id, args.region)
+    write_image_to_csv(args.image_id, args.region)
+
+    # Process other regions
+    csv_lock = threading.Lock()
+
+    def process_region(copy_to_region):
+        print(f"Start copying image to {copy_to_region}...")
+        try:
+            new_image_id = copy_image_and_make_public(copy_to_region)
+        except Exception as e:
+            print(f"Error generating image to {copy_to_region}: {str(e)}")
+            new_image_id = 'NEED_FALLBACK'
+        with csv_lock:
+            write_image_to_csv(new_image_id, copy_to_region)
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        executor.map(process_region, ALL_REGIONS)
+    executor.shutdown(wait=True)
```

Comment:
It might be cleaner to save the rows in memory and write to file all at once at the end, so that we can sort those lines before writing into the file. Reason: we may want to sort by the regions, as that will make it easier to track changes to the images.csv